src/utilities/separateOperations.js

```diff
@@ -47,23 +52,32 @@ export function separateOperations(
   // For each operation, produce a new synthesized AST which includes only what
   // is necessary for completing that operation.
   const separatedDocumentASTs = Object.create(null);
-  operations.forEach(operation => {
-    const operationName = opName(operation);
+  const operationNames = Object.keys(definitions).filter(defName =>
```

Comment:
Leaving the existing list of operations as it was would remove the need to do a second pass through all the definitions